timlee/networking-with-a-raspberry/README.md

```diff
@@ -0,0 +1,280 @@
+
+![Servers](./servers.png)
+
+### Server options
+
+So which server software to use? There are many popular options including<sup>[1](#1)</sup>:
+
+- **[Apache](https://httpd.apache.org/)** - open source, around since 1995, serves around 25% of the web
+- **[Nginx](https://www.nginx.com/)** - open source, released in 2004, more performant than Apache and has been growing in popularity, stealing a significant market share from Apache, serves around 37% of the web
+- **[IIS](https://www.iis.net/)** (Internet Information Services) - created by Microsoft, tightly coupled to Windows, serves around 13% of the web
+
+For this demo, we will be using Nginx because it's easy to set-up and incredibly flexible, also serving as a reverse proxy (sneak peak of what's to come).
+
+### Setting up Nginx
+
+The following terminal commands on your Raspberry Pi should do the trick:
+
+- `sudo apt update` to ensure that the package manager is up-to-date
+- `sudo apt install nginx` to install nginx
+- `sudo /etc/init.d/nginx start` to start the server
+
+You can then go to `http://localhost` on the Pi and see the default `Welome to ngingx!` page!
+
+And if you find the IP address of your Raspberry Pi (e.g. by typing `ifconfig` in the terminal) then you should be able to access it from your own machine by going to that IP in the browser.
+
+The base configuration file for the server can be found at `/etc/nginx/nginx.conf`.
+
+The settings for serving the default site on localhost are within the file `/etc/nginx/sites-available/default`, which establishes a single server, listening for all requests on port 80 (HTTP) and serving the HTML file `/var/www/html/index.nginx-debian.html`. This file is the entry point for your site.
+
+## Step 2: DNS
+
+So we can now access our site from any device on the network by using the IP address of our Raspberry Pi. But an IP address isn't the most ergonomic way to reach a site - all of our keen badger enthusiasts aren't going to want to type that in every time they want to get some badger facts. Better would be a memorable name, like... `badger.facts`, right? So how do we make that happen?
+
+We need to provide a mechanism to convert that memorable name that a user enters into an IP address so their computers can know where to go. We need DNS.
+
+### What is DNS?
+
+The Domain Name System is the address book of the internet, making sure each browser doesn't get lost. A huge web of dedicated, interlinked DNS servers help answer every query on the internet, receiving requests like "badgerland.com?" and responding with "47.43.22.52".
```

Comment:
Maybe introduce a link to DNS Wikipedia article https://en.wikipedia.org/wiki/Domain_Name_System